homeassistant/components/airzone/config_flow.py

```diff
@@ -76,3 +85,78 @@ async def async_step_user(
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP discovery."""
+        self._discovered_ip = discovery_info.ip
+        self._discovered_mac = discovery_info.macaddress
+
+        _LOGGER.debug(
+            "DHCP discovery detected Airzone WebServer: %s", self._discovered_mac
+        )
+
+        options = ConnectionOptions(self._discovered_ip)
+        airzone = AirzoneLocalApi(
+            aiohttp_client.async_get_clientsession(self.hass), options
+        )
+        try:
+            await airzone.get_version()
+        except AirzoneError as err:
+            raise AbortFlow("cannot_connect") from err
+
+        self._async_abort_entries_match({CONF_HOST: self._discovered_ip})
+
+        await self.async_set_unique_id(format_mac(self._discovered_mac))
+        self._abort_if_unique_id_configured()
```

Comment:
Move this above the connection attempt.  We shouldn't try to connect if we are going to abort anyways

Reply:
@bdraco done in f23e96d1ffb0b4479034621aa2b4cbdb3be6a237